modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java

```diff
@@ -27,55 +27,54 @@
 import java.util.Date;
 
 import org.osgi.framework.BundleContext;
-import org.osgi.service.component.annotations.Activate;
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Andrew Betts
  */
-@Component(
-	property = "background.task.executor.class.name=com.liferay.portal.search.internal.background.task.ReindexSingleIndexerBackgroundTaskExecutor",
-	service = {
-		BackgroundTaskExecutor.class,
-		ReindexSingleIndexerBackgroundTaskExecutor.class
-	}
-)
 public class ReindexSingleIndexerBackgroundTaskExecutor
 	extends BaseReindexBackgroundTaskExecutor {
 
+	public ReindexSingleIndexerBackgroundTaskExecutor(
+		IndexerRegistry indexerRegistry,
+		ReindexStatusMessageSender reindexStatusMessageSender,
+		SearchEngineHelper searchEngineHelper,
+		ServiceTrackerList<Indexer<?>> systemIndexers) {
+
+		_indexerRegistry = indexerRegistry;
+		_reindexStatusMessageSender = reindexStatusMessageSender;
+		_searchEngineHelper = searchEngineHelper;
+		_systemIndexers = systemIndexers;
```

Comment:
@rcsdnj If this class has a service tracker list internally, there's no need to pass one from the constructor.

Reply:
@rcsdnj After a second thought, why do we turn the component into a non-component? Can we just remove the `ReindexSingleIndexerBackgroundTaskExecutor` from `BackgroundTaskExecutorConfigurator`?

Maybe someday we need to clean up the self-activating-service-as-a-configurator pattern.

Reply:
@rafaprax What do you think?

Reply:
Hey @dantewang,

That sounds good to me. @rcsdnj could you please use the approach mentioned by Dante?

> Maybe someday we need to clean up the self-activating-service-as-a-configurator pattern.

Yeah, If the priorities do not change, It will be the next task we will work on.

Thx
Thx